src/main/java/Duke.java

```diff
@@ -1,10 +1,265 @@
+
+    private Storage storage;
+    private TaskList taskList;
+
+    /**
+     * Constructs a new instance of Duke.
+     *
+     * @param taskList the task list
+     */
+    public Duke(TaskList taskList) {
+
+        storage = new Storage(Paths.get(System.getProperty("user.home"), "data", "duke.txt"));
+        this.taskList = taskList;
+        storage.loadTasks(taskList);
     }
+
+    /**
+     * Gets Duke's task list.
+     *
+     * @return Duke's task list.
+     */
+    public TaskList getTaskList() {
+
+        return taskList;
+    }
+
+    /**
+     * Executes the command.
+     *
+     * @param command the string command supplied by the user.
+     * @return the relevant response after processing the command.
+     */
+    public Response executeCommand(String command) {
+
+        try {
+            //Split the command tokens into description and time portions
+            String[] commandTokens = Parser.parse(command, "/");
+
+            //Further split description into words for command checking
+            String[] descriptionTokens = Parser.parse(commandTokens[0], " ");
+
+            if (commandTokens.length < 2) { //Regular commands and todo
+
+                switch (descriptionTokens[0].toLowerCase()) {
+                case "bye":
+                    storage.saveTasks(taskList);
+                    return new Response(Message.GOODBYE, null);
+
+                case "delete":
+                    //Check if there are sufficient tokens
+                    checkSufficientTokens(descriptionTokens, "delete");
+
+                    int index = Integer.parseInt(descriptionTokens[1]) - 1;
+                    return new Response(Message.DELETE, taskList.remove(index));
+
+                case "done":
+                    //Check if there are sufficient tokens
+                    checkSufficientTokens(descriptionTokens, "done");
+
+                    int indx = Integer.parseInt(descriptionTokens[1]) - 1;
+                    taskList.complete(indx);
+                    return new Response(Message.DONE, indx);
+
+                case "find":
+                    //Check if there are sufficient tokens
+                    checkSufficientTokens(descriptionTokens, "find");
+
+                    TaskList results = taskList.list(descriptionTokens[1].toLowerCase());
+                    if (results.getSize() > 0) {
+                        return new Response(Message.TASK_FOUND, results);
+                    } else {
+                        return new Response(Message.TASK_NOT_FOUND, null);
+                    }
+
+                case "list":
+                    if (taskList.getSize() > 0) {
+                        if (descriptionTokens.length == 2) {
+                            //Date is provided
+                            LocalDate date = LocalDate.parse(descriptionTokens[1]);
+                            TaskList tmp = taskList.list(date);
+                            if (tmp.getSize() > 0) {
```

Comment:
Nested if statements are not recommended as they are very difficult to understand. If possible think of a possible way to avoid too many nested if conditions.